api/v1beta1/openstackprovisionserver_types.go

```diff
@@ -53,7 +54,10 @@ const (
 // OpenStackProvisionServerSpec defines the desired state of OpenStackProvisionServer
 type OpenStackProvisionServerSpec struct {
 	// Port - The port on which the Apache server should listen
-	Port int32 `json:"port"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=6190
+	// +kubebuilder:validation:Maximum=6220
```

Comment:
I wonder if this could cause RHOSO version updates to fail if the user currently has `OpenStackProvisionServer`s deployed outside of this range.  We've seen this lately with certain upgrades where a CRD can't be updated because it would violate what is currently deployed, and so the whole upgrade and install of the new version is blocked.

Reply:
That's only for pre-created provision servers right? Can't those  updated before update (if there are any)? We can document that. My concern is with allowing users to use any port for the provision server.

Reply:
Yes, it would be for existing CRs.  And the workaround is just to update them.  So we might need to document that.  I didn't mean to imply that we shouldn't make this change, it was just a heads-up.

Reply:
> Yes, it would be for existing CRs

Probably not for all existing CRs, only the ones pre created with a port higher than the range(?). Do we have any customer using RHOSO in production yet? The only one I know of that tried to use pre-created provision server is not in production yet.

Reply:
Sorry, I didn't mean all existing CRs would have a problem, just that they'd all be considered against the new constraint.  After thinking about my initial comment, I don't think there's much to worry about here.  Documenting the restriction as a heads-up might be prudent, but users can always just delete existing `OpenStackProvisionServer`s if there's a problem during upgrade.  They're intended to be ephemeral and thus there's an easy workaround in just deleting the problematic ones.  I guess unless the user _really_ needed to use a port that was outside the new rage for some reason, but that seems unlikely.

Reply:
@abays I guess we can merge this and then update the downstream documentation in FR2 ?

Reply:
Sure, sounds reasonable